src/transformers/pipelines/text_generation.py

```diff
@@ -20,11 +20,24 @@ class ReturnType(enum.Enum):
     FULL_TEXT = 2
 
 
+class Chat:
+    """This class is intended to just be used internally in this pipeline and not exposed to users. We convert chats
+    to this format because the rest of the pipeline code tends to assume that lists of messages are
+    actually a batch of samples rather than messages in the same conversation."""
+
+    def __init__(self, messages):
```

Comment:
```suggestion
    def __init__(self, messages: dict):
```
nit: helps understanding that a dict is expected

Reply:
Done!